core/domain/blog_services.py

```diff
@@ -116,7 +158,27 @@ def get_blog_post_summary_from_model(blog_post_summary_model):
         blog_post_summary_model.published_on)
 
 
-def get_blog_post_summary_by_id(blog_post_id, strict=False):
+@overload
+def get_blog_post_summary_by_id(
+    blog_post_id: str
+) -> blog_domain.BlogPostSummary: ...
+
+
+@overload
+def get_blog_post_summary_by_id(
+    blog_post_id: str, *, strict: Literal[True]
+) -> blog_domain.BlogPostSummary: ...
+
+
+@overload
+def get_blog_post_summary_by_id(
+    blog_post_id: str, *, strict: Literal[False]
+) -> Optional[blog_domain.BlogPostSummary]: ...
+
+
+def get_blog_post_summary_by_id(
+    blog_post_id: str, strict: bool = True
+) -> Optional[blog_domain.BlogPostSummary]:
```

Comment:
I have not used  `*` in the main implementation, because doing this throws lint errors and those lint errors are showing up because after using single `*` `strict` is not counted as parameter and its documentation is just considered as a description of `blog_post_id` variable.

If we use `*args` then linter ask for it's documentation. 
Also, not using `*` in the main implementation does not affect the expected behaviour because If we use `get(aaa, True)` then MyPy ultimately shows an error.